### PR TITLE
[INLONG-6822][Sort] Add multi table source for MongoDb

### LINKED ISSUE #6822: [Feature][Sort]Add multi table source for MongoDb
### Description

Add multi table source for MongoDb

### Use case

Add multi table source for MongoDb

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kuansix**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-6822] Add multi table source for MongoDb

- Fixes #6822 

### Motivation

 - Users want to real time sync multi-tables data in mongo to pg、kafka etc in many case. We need multi table-read connector for MongoDb.

 - Based on debezium-connector-mongodb, mongodb-cdc-inlong can read data and assemble two data formats: canal-json and debezium-json.

### Modifications

1. MongoDBConnectorDeserializationSchema's invoke method supports handling debezium canal-json upstream parsing. It picks data from upstream, assembles data with json format.
2. RowKindValidator's set only specified row kinds can be send to sink.
3. RecordUtils's convert data of MongoDb BsonType to flink LogicalType.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Comment by yunqingmoswu** [comment]
Please add describe for your design and implements.

File: inlong-sort/sort-connectors/mongodb-cdc/pom.xml [hunk 1]
```diff
@@ -44,6 +50,31 @@
             <artifactId>audit-sdk</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-table-common</artifactId>
+            <version>1.13.5</version>
+        </dependency>
```
Comment (gong) [inline inlong-sort/sort-connectors/mongodb-cdc/pom.xml:57]: no need it

File: inlong-sort/sort-connectors/mongodb-cdc/pom.xml [hunk 2]
```diff
@@ -44,6 +50,31 @@
...
+        <dependency>
+            <groupId>io.debezium</groupId>
+            <artifactId>debezium-connector-oracle</artifactId>
+            <version>${debezium.version}</version>
+        </dependency>
```
Comment (gong) [inline inlong-sort/sort-connectors/mongodb-cdc/pom.xml:62]: remove it, error dependency

File: inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java [hunk 3]
```diff
@@ -493,15 +492,6 @@ public void deserialize(SourceRecord record, Collector<T> out) throws Exception
...
 
-                            @Override
-                            public void deserialize(SourceRecord record, Collector<T> out,
-                                    TableChanges.TableChange tableSchema) throws Exception {
-                                if (sourceMetricData != null) {
-                                    sourceMetricData.outputMetricsWithEstimate(record.value());
-                                }
-                                deserializer.deserialize(record, out, tableSchema);
-                            }
-
```
Comment (gong) [inline inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java]: why remove it

File: inlong-sort/sort-connectors/mongodb-cdc/pom.xml [hunk 4]
```diff
@@ -74,6 +105,7 @@
                                     <include>org.apache.avro:avro</include>
                                     <include>org.apache.kafka:*</include>
                                     <include>com.fasterxml.*:*</include>
+                                    <include>com.google.code.gson:*</include>
```
Comment (gong) [inline inlong-sort/sort-connectors/mongodb-cdc/pom.xml:108]: no need it

Comment (kuansix) [reply inlong-sort/sort-connectors/mongodb-cdc/pom.xml:108]: MongoDBConnectorDeserializationSchema.java L720 need

File: inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java (hunk 3 above)
Comment (kuansix) [reply inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java]: > why remove it

mongo has no schema

File: inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/debezium/table/MongoDBConnectorDeserializationSchema.java [hunk 5]
```diff
@@ -0,0 +1,876 @@
...
+                    default:
+                        valueStr = valueStr + bsonValue + ",";
+                }
+            }
+            valueStr = valueStr.substring(0, valueStr.length() - 1) + "}";
+            return StringData.fromString(valueStr);
+        }
+        if (docObj.isDocument()) {
+            return StringData.fromString(gson.toJson(docObj));
+        }
```
[the remark this answers is not in the document]
Comment (kuansix) [reply inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/debezium/table/MongoDBConnectorDeserializationSchema.java:721]: > gson can be replaced by jackson. flink has use jackson
jackson occur error in many case

**Review by yunqingmoswu — APPROVED** [review]

**Review by e-mhui — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Comment by gong** [comment]
use mvn spotless:apply

**Comment by kuansix** [comment]
> use mvn spotless:apply

done

**Comment by kuansix** [comment]
> Please add describe for your design and implements.

done

**Review by EMsnap — APPROVED** [review]

**Pull request merged**